rdt/hyper_transformer.py

```diff
@@ -417,14 +469,15 @@ def _learn_config(self, data):
         """Unfit the HyperTransformer and learn the sdtypes and transformers of the data."""
         self._unfit()
         for field in data:
-            if field not in self.field_sdtypes:
+            if field not in self.config['field_sdtypes']:
                 self._set_field_sdtype(data, field)
-            if field not in self.field_transformers:
-                sdtype = self.field_sdtypes[field]
+            if field not in self.config['field_transformers']:
+                sdtype = self.config['field_sdtypes'][field]
                 if sdtype in self._default_sdtype_transformers:
-                    self.field_transformers[field] = self._default_sdtype_transformers[sdtype]
+                    default_transformer = self._default_sdtype_transformers[sdtype]
+                    self.config['field_transformers'][field] = default_transformer
```

Comment:
I was thinking of adding methods to let the `HyperTransformer` interact with the `Config` instead of doing it directly. That way people don't accidently forget to update the `_provided` dicts if necessary. So maybe an `add_transformer(provided)` method that adds it to `config.field_transformers` and if `provided` is `True`, then also `config._provided_transformers`. Same for `field_sdtypes`